src/mbgl/style/style_bucket.hpp

```diff
@@ -91,18 +101,20 @@ class StyleBucket {
     typedef util::ptr<StyleBucket> Ptr;
 
     StyleBucket(StyleLayerType type);
-
+    StyleLayerType type;
     std::string name;
     util::ptr<StyleSource> style_source;
     std::string source_layer;
     FilterExpression filter;
+    ClassProperties layout;
     StyleBucketRender render = std::false_type();
     float min_zoom = -std::numeric_limits<float>::infinity();
     float max_zoom = std::numeric_limits<float>::infinity();
     VisibilityType visibility = VisibilityType::Visible;
```

Comment:
@jfirebaugh good point — there was some trickery here since layout properties became parsed differently; about to push a fix that removes all individual `StyleBucket*` `visibility` properties.